pkg/pair/match_test.go

```diff
@@ -0,0 +1,136 @@
+package pair
+
+import (
+	"bytes"
+	"context"
+	"encoding/csv"
+	"fmt"
+	"optable-pair-cli/pkg/io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMatch(t *testing.T) {
+	t.Parallel()
+
+	// arrange
+	lenEmails := 1001
+	ctx := context.Background()
+	salt := requireGenSalt(t)
+	key1, key2 := requireGenKey(t), requireGenKey(t)
+	emails := requireGenRandomHashedEmails(t, lenEmails)
+	encryptedEmails1 := requireEncryptEmails(t, emails[:900], salt, key1)
+	encryptedEmails2 := requireEncryptEmails(t, emails[100:], salt, key2)
+	twiceEncryptedEmails1 := requireReEncryptEmails(t, encryptedEmails1, salt, key2)
+	twiceEncryptedEmails2 := requireReEncryptEmails(t, encryptedEmails2, salt, key1)
+	r1, r2 := bytes.NewBuffer(nil), bytes.NewBuffer(nil)
+
+	requireWriteEmails(t, r1, twiceEncryptedEmails1)
+	requireWriteEmails(t, r2, twiceEncryptedEmails2)
+
+	// create map to access encrypted emails faster
+	expectContain := make(map[string]struct{}, 1000)
```

Comment:
```suggestion
	expectContain := make(map[string]struct{}, 901)
```

It should be exactly 901 emails matched, no?